### ISSUE TITLE
[FEATURE] : add pagination support for customers list

### ISSUE THREAD

**Issue by Narayanbhat166** [juspay/hyperswitch#3746]
### Feature Description

The [customers list](https://github.com/juspay/hyperswitch/blob/4ae28e48cd73a9f96b6ae24babf167824fd182a0/crates/router/src/routes/customers.rs#L68) endpoint currently has no pagination support. It returns the list of all the customers in the database. There would be thousands of customers for a merchant. A pagination support for the api would be necessary.

### Possible Implementation

accept two fields, `limit` and `offset`
- `limit` specifies the number of objects returned in the response.
- `offset` specifies the number of items to skip from the start, this is open to discussion whether to use offset or some other field.

This would be similar to the `payments_list` pagination. Refer [here](https://github.com/juspay/hyperswitch/blob/54c0b218d9f20c6388dde0f30ea525011cdad573/crates/router/src/routes/payments.rs#L868)

### Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

None

**Comment by tanbirali** [comment]
I am interested to take this issue, I will like to know little bit more, from what I saw both product list and customer list look same to me.

**Comment by Narayanbhat166** [comment]
Hey @tanbirali, thanks for showing interest. Payment list accepts `limit` and [other parameters](https://github.com/juspay/hyperswitch/blob/main/crates/api_models/src/payments.rs#L2531), which are used for pagination. These parameters are not available in customers list endpoint.

**Comment by tanbirali** [comment]
Okay, correct me if I am wrong, I have to create a CustomerListConstraints where I need to add limit and offset

**Comment by Narayanbhat166** [comment]
Yes, you would have to create the struct and add limit and offset to it.

**Comment by hegdeadithyak** [comment]
Hi there @Narayanbhat166 ,

I'd like to take a look at this issue and see if I can help fix it. Please let me know if it's already assigned to someone or if there's anything specific I should keep in mind while working on it.

Thanks,
**Adithya Hegde Kota**

**Comment by JamesM25** [comment]
Hi @Narayanbhat166, I noticed this issue hasn't had any activity recently. I am interested in working on it.

**Comment by Narayanbhat166** [comment]
Hey @hegdeadithyak, let me know if you are interested in contributing. I can get you another issue to work on

**Comment by hegdeadithyak** [comment]
> Hey @hegdeadithyak, let me know if you are interested in contributing. I can get you another issue to work on

Yes, I am Intrested.

Thanks,
**Adithya Hegde Kota**
